ddtrace/contrib/aioredis/patch.py

```diff
@@ -0,0 +1,74 @@
+import aioredis
+
+from ddtrace import config
+from ddtrace.internal.utils.wrappers import unwrap as _u
+from ddtrace.pin import Pin
+from ddtrace.vendor.wrapt import wrap_function_wrapper as _w
+
+from ..redis.util import _trace_redis_cmd
+from ..redis.util import _trace_redis_execute_pipeline
+from ..redis.util import format_command_args
+
+
+config._add("aioredis", dict(_default_service="redis"))
+
+aioredis_version_str = getattr(aioredis, "__version__", "0.0.0")
+aioredis_version = tuple([int(i) for i in aioredis_version_str.split(".")])
+
+
+def patch():
+    if getattr(aioredis, "_datadog_patch", False):
+        return
+    setattr(aioredis, "_datadog_patch", True)
+    pin = Pin()
+    if aioredis_version >= (2, 0):
+        _w("aioredis.client", "Redis.execute_command", traced_execute_command)
+        _w("aioredis.client", "Redis.pipeline", traced_pipeline)
+        _w("aioredis.client", "Pipeline.execute", traced_execute_pipeline)
+        pin.onto(aioredis.client.Redis)
+    else:
+        _w("aioredis", "Redis.execute", traced_execute_command)
```

Comment:
This is the wrong way to patch aioredis 1.3, this will cause pipelines to not work as expected, and we cannot use our async wrappers here since `Redis.execute` in 1.3 is not an async function, instead it manually creates and returns as task which means callers are expecting to get back a task (and trying to do things like call `fut.add_done_callback`) and we are calling `await` for them.

We either need to remove < 2.0 support, or we need to revisit how we are patching (and testing) support for 1.3.x.